packages/firestore/src/local/indexeddb_remote_document_cache.ts

```diff
@@ -172,18 +167,26 @@ export class IndexedDbRemoteDocumentCache implements RemoteDocumentCache {
     let changedDocs = maybeDocumentMap();
 
     const range = IDBKeyRange.lowerBound(
-      this._lastProcessedDocumentChangeId,
-      /*lowerOpen=*/ true
+      this._lastProcessedDocumentChangeId + 1
     );
+    let firstIteration = true;
 
-    // TODO(b/114228464): Another client may have garbage collected the remote
-    // document changelog if our client was throttled for more than 30 minutes.
-    // We can detect this if the `lastProcessedDocumentChangeId` entry is no
-    // longer in the changelog. It is possible to recover from this state,
-    // either by replaying the entire remote document cache or by re-executing
-    // all queries against the local store.
     return documentChangesStore(transaction)
       .iterate({ range }, (_, documentChange) => {
+        if (firstIteration) {
+          // If our client was throttled for more than 30 minutes, another
+          // client may have garbage collected the remote document changelog.
+          if (this._lastProcessedDocumentChangeId + 1 !== documentChange.id) {
+            return PersistencePromise.reject(
+              new FirestoreError(
+                Code.DATA_LOSS,
+                REMOTE_DOCUMENT_CHANGELOG_GARBAGE_COLLECTED_ERR_MSG
+              )
+            );
+          }
+          firstIteration = false;
```

Comment:
Per my comment elsewhere, perhaps rework this so that getNewDocumentChanges() always advances past all document changes, even if it ultimately returns an error...  e.g. by changing firstIteration to minChangeId and checking it in the next() block...

Reply:
Consider it done.